api/server/router/network/network_routes.go

```diff
@@ -80,7 +80,7 @@ func (n *networkRouter) postNetworkCreate(ctx context.Context, w http.ResponseWr
 		return err
 	}
 
-	if _, err := n.clusterProvider.GetNetwork(create.Name); err == nil {
+	if _, err := n.clusterProvider.GetNetworkByName(create.Name); err == nil {
 		return libnetwork.NetworkNameError(create.Name)
```

Comment:
Not directly related to this PR, but wondering; should this check only be performed if `create.CheckDuplicate` is true? (see [daemon/network.go#L256-L258](https://github.com/docker/docker/blob/4fae94f6637bfbbeba809f930f2ecf1e36748882/daemon/network.go#L256-L258))

I see this was added in [#24158](https://github.com/docker/docker/pull/24158/commits/0ce5158a2a9a3f10a62d3c1ea289c55e524cdac5#diff-6e937312ed7fe823e2b7fe0d1215fb60R84)

Since allowing duplicate names is by design (https://github.com/docker/docker/issues/18864#issuecomment-167006094), I wonder if it should error out.

(t.b.h. I would rather stop allowing duplicate names, because I think the _name_ is used in many cases as the only reference)

ping @mavenugo ^^